pkg/runtime/wfengine/workflow.go

```diff
@@ -331,19 +328,28 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 		} else {
 			continue
 		}
-		req := actors.TransactionalRequest{
-			ActorType: wf.config.activityActorType,
-			ActorID:   getActivityActorID(actorID, taskID, state.Generation),
-			Operations: []actors.TransactionalOperation{{
-				Operation: actors.Delete,
-				Request: actors.TransactionalDelete{
-					Key: activityStateKey,
-				},
-			}},
+		op := actors.TransactionalOperation{
+			Operation: actors.Delete,
+			Request: actors.TransactionalDelete{
+				Key: activityStateKey,
+			},
 		}
-		err = wf.actors.TransactionalStateOperation(ctx, &req)
+		activityActorID := getActivityActorID(actorID, taskID, state.Generation)
+		if transactionalRequests[activityActorID] == nil {
+			transactionalRequests[activityActorID] = []actors.TransactionalOperation{op}
+		} else {
+			transactionalRequests[activityActorID] = append(transactionalRequests[activityActorID], op)
+		}
+	}
+	// TODO: for optimization make multiple go routines and run them in parallel
+	for activityActorID, operations := range transactionalRequests {
+		err = wf.actors.TransactionalStateOperation(ctx, &actors.TransactionalRequest{
```

Comment:
With this change to put all the deletes into a single transaction, do we run the risk of exceeding Cosmos DB transaction batch size limitations?

Reply:
This doesn't put _all_ deletes in a single transaction, but only those for the same actor (because of partition keys). But yes, we do.

- We have been talking about dropping support for Cosmos DB anyways, since it has issues also when saving state
- If absolutely needed, we can split this into multiple calls to TransactionalStateOperation. Since these are just deletes, it's _probably_ ok if we don't have atomicity here.